### PR TITLE
Add codeowners, issue template and pr template

### LINKED ISSUE #173: We should have issue and pr templates
This would make it easier for us to communicate better during the software development process and set good expectations on the level of detail we should be providing as we go.

### PR DESCRIPTION
**Pull request by evansiroky**
### Checklist

- [x] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
- [x] The description lists all applicable issues this PR seeks to resolve
- [x] The description lists any configuration setting(s) that need to be enabled/disabled in order for the software to run properly with these changes
- [x] All tests and CI builds passing

### Description

This PR adds codeowners, an issue template and a PR tempalte in order to resolve #173.

### CODE REVIEW

File: .github/issue_template.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+## Steps to reproduce the problem
+
+Please be as specific as possible.
+
+## Any special notes on configuration used
+
+Please describe any applicable config files that were used
+
+## Version of datatools-server (exact commit hash or branch name)
```
Comment (landonreed) [inline .github/issue_template.md]: Perhaps we should optionally have issue creators include the datatools-ui commit hash/branch name if applicable (or maybe always).

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
+### Checklist
+
+- [ ] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
+- [ ] The description lists all applicable issues this PR seeks to resolve
+- [ ] The description lists any configuration setting(s) that need to be enabled/disabled in order for the software to run properly with these changes
```
Comment (landonreed) [inline .github/pull_request_template.md]: Maybe we should clarify this point to say "any configuration setting(s) that differ from the default settings."

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -0,0 +1,12 @@
+### Checklist
+
+- [ ] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
```
Comment (landonreed) [inline .github/pull_request_template.md]: We should add a checklist item that says something like "Any modified or new methods or classes have helpful Javadoc and code is thoroughly commented." Maybe we can link to a commenting style that we like.

File: .github/pull_request_template.md [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+- [ ] The description lists all applicable issues this PR seeks to resolve
+- [ ] The description lists any configuration setting(s) that need to be enabled/disabled in order for the software to run properly with these changes
+- [ ] All tests and CI builds passing
+- [ ] e2e tests are all passing _(remove this if not merging to master)_
+- [ ] Code coverage improves or is at 100% _(remove this if not merging to master)_
+
+### Description
+
+Please explain the changes you made here. Be sure to include all issues being resolved and any special configuration settings that are need for the software to run properly with these changes.
```
Comment (landonreed) [inline .github/pull_request_template.md]: Maybe expand this first sentence to say:

```
Please explain the changes you made here and, if not immediately obvious from the code, how they resolve any referenced issues.
```

File: .github/issue_template.md [hunk 5]
```diff
@@ -0,0 +1,21 @@
+_**NOTE:** this issue system is intended for reporting bugs and tracking progress in software development. Although this software is licensed with an open-source license, any issue opened here may not be responded to in a timely manner. [Conveyal](https://www.conveyal.com) is unable to provide technical support for custom deployments of this software unless your company has a support contract with us. Please remove this note when creating the issue._
```
Comment (landonreed) [inline .github/issue_template.md]: Beautiful.

File: .github/pull_request_template.md (hunk 3 above)
Comment (evansiroky) [reply .github/pull_request_template.md]: Yeah a style guide would definitely be helpful. I'll just add a note about javadoc for now.

**Comment by landonreed** [comment]
Good stuff, thanks @evansiroky.

**Pull request merged**